auth/password_authenticator.cc

```diff
@@ -149,13 +149,15 @@ future<> password_authenticator::create_default_if_missing() const {
     });
 }
 
-future<> password_authenticator::start(service&) {
-     return once_among_shards([this] {
+future<> password_authenticator::start(service& svc) {
+     return once_among_shards([this, &svc] {
          auto f = create_metadata_table_if_missing(
                  meta::roles_table::name,
                  _qp,
                  meta::roles_table::creation_query(),
-                 _migration_manager);
+                 _migration_manager).then([&svc] {
+                     svc.add_protector([] (command_desc) { return true; });
```

Comment:
Maybe add_protector should belong to authorizer, not auth::service.

Reply:
I don't see it that way.  `client_state` relies on a `service` member to determine access rights:
https://github.com/scylladb/scylla/blob/fcec68b1028672cffe0491e2672c1b927e79cc57/service/client_state.cc#L191

And `service`, in turn treats authorizer as just one of its clients declaring protected assets:
https://github.com/scylladb/scylla/blob/fcec68b1028672cffe0491e2672c1b927e79cc57/auth/service.cc#L374-L378

Reply:
Well, it's already this way so we can keep it. But in general we should disaggregate large blobs.

authenticator is unneeded by cql and needed by transport
authorizer is needed by cql and unneeded by transport

If each gets just what it needs, recompiles are fewer.